src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -7852,11 +7871,24 @@ void StorageReplicatedMergeTree::replacePartitionFrom(
     ProfileEventsScope profile_events_scope;
 
     MergeTreeData & src_data = checkStructureAndGetMergeTreeData(source_table, source_metadata_snapshot, metadata_snapshot);
-    String partition_id = getPartitionIDFromQuery(partition, query_context);
+    String partition_id = src_data.getPartitionIDFromQuery(partition, query_context);
```

Comment:
@tavplubix 

On regular MergeTree, the new `partition_id` is only used when it is time to create the new data part. Therefore, it is computed inside `cloneAndLoadPartOnSameDiskWithDifferentPartitionKey`.

In this case (Replicated), it is used to allocate blocks on zookeeper. I think this actually needs to use the updated partition id